test/server/http_filter_integration_test_base.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <string>
+
+#include "external/envoy/test/integration/http_integration.h"
+
+namespace Nighthawk {
+
+/**
+ * Base class with shared functionality for testing Nighthawk test server http filter extensions.
```

Comment:
Can we add two notes/clarifications here: 1. This class is stateful.  2. Unless I'm mistaken I don't think this class is threadsafe